Source/Core/AudioCommon/Mixer.cpp

```diff
@@ -51,134 +105,146 @@ unsigned int CMixer::MixerFifo::Mix(short* samples, unsigned int numSamples, boo
 		aid_sample_rate = aid_sample_rate * (framelimit - 1) * 5 / VideoInterface::TargetRefreshRate;
 	}
 
-	const u32 ratio = (u32)( 65536.0f * aid_sample_rate / (float)m_mixer->m_sampleRate );
+	// ratio = 1 / upscale_factor = stepsize for each sample
+	// e.g. going from 32khz to 48khz is 1 / (3 / 2) = 2 / 3
+	// note because of syncing and framelimit, ratio will rarely be exactly 2 / 3
+	float ratio = aid_sample_rate / (float) m_mixer->m_sample_rate;
+
+	float l_volume = (float) m_lvolume / 255.f;
+	float r_volume = (float) m_rvolume / 255.f;
+
+	// for each output sample pair (left and right),
+	//   linear interpolate between current and next sample
+	//   increment output sample position
+	//   increment input sample position by ratio, store fraction
+	// QUESTION: do we need to check for NUM_CROSSINGS samples before we interpolate?
```

Comment:
Can someone look at this and let me know if I need to check write_index - read_index > NUM_CROSSINGS